### PR TITLE
added logging for every google call

### PR DESCRIPTION
**Pull request by dvoet**
https://broadinstitute.atlassian.net/browse/DSDEEPB-2637
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/rawls/integrationtest/HttpGoogleServicesDAOSpec.scala [hunk 1]
```diff
@@ -94,6 +94,10 @@ class HttpGoogleServicesDAOSpec extends FlatSpec with Matchers with IntegrationT
     val ownerResource = Await.result(retry(when500)(() => Future { directory.groups.get(ownerGroup.groupEmail.value).execute() }), Duration.Inf)
 
     // delete the workspace bucket and groups. confirm that the corresponding groups are deleted
+    mockDataSource.inTransaction() { txn =>
+      VertexSchema.createVertexClasses(txn.graph)
+      containerDAO.workspaceDAO.savePendingBucketDeletions(PendingBucketDeletions(Set.empty), txn)
+    }
```
Comment (dvoet) [inline src/test/scala/org/broadinstitute/dsde/rawls/integrationtest/HttpGoogleServicesDAOSpec.scala:100]: the above makes the call to deleteWorkspace actually try to remove the bucket

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala [hunk 2]
```diff
@@ -531,4 +534,32 @@ class HttpGoogleServicesDAO(
...
+        request.executeUnparsed()
+      } match {
+        case Success(response) =>
+          logger.debug(s"GOOGLE REQUEST: ${request.getRequestMethod} ${request.buildHttpRequestUrl()}, payload: ${payload}, SUCCESS http status code: ${response.getStatusCode}, time ${System.currentTimeMillis() - start} ms")
+          response.parseAs(request.getResponseClass)
+        case Failure(httpRegrets: HttpResponseException) =>
+          logger.debug(s"GOOGLE REQUEST: ${request.getRequestMethod} ${request.buildHttpRequestUrl()}, payload: ${payload}, FAILED http status code: ${httpRegrets.getStatusCode}, time ${System.currentTimeMillis() - start} ms")
+          throw httpRegrets
+        case Failure(regrets) =>
+          logger.debug(s"GOOGLE REQUEST: ${request.getRequestMethod} ${request.buildHttpRequestUrl()}, payload: ${payload}, EXCEPTION message: ${regrets.getMessage}, time ${System.currentTimeMillis() - start} ms")
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala]: Idea: can (should?) we make these JSON?

(might be a dumb idea, requesting feedback)

**Comment by helgridly** [comment]
:+1:

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala (hunk 2 above)
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala]: sure, why not... maybe someone can parse it

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala]: In hindsight this seems silly. Let's not.

Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala]: Vomiting JSON into logfiles is eww. I'm sure the regex for this isn't difficult to write.

**Comment by helgridly** [comment]
sure, still :+1:

**Pull request merged**